doc/coq.rst

```diff
@@ -0,0 +1,392 @@
+.. _coq:
+
+***
+Coq
+***
+
+.. contents:: Table of Contents
+    :depth: 3
+
+Introduction
+------------
+
+Dune can build Coq theories and plugins with additional support for extraction
+and ``.mlg`` file preprocessing.
+
+A *Coq theory* is a collection of ``.v`` files all prefixed by a common Coq
+module prefix, following the directory hierarchy for module names.
```

Comment:
```suggestion
A *Coq theory* is a collection of ``.v`` files that generate Coq modules whose
names that share a common prefix.  The module names reflect the directory
hierarchy.
```

Reply:
I think that writing ".v files generate modules" is kinda misleading. .v files are modules (special ones tho)